### ISSUE TITLE
Ignore trailing comments in `newline-after-var` rule

### ISSUE THREAD

**Issue by cvisco** [eslint/eslint#2229]
The `newline-after-var` rule is issuing a warning in cases where there may be a trailing comment after the `var` declaration:

The following case is currently considered a warning, even though a newline follows the var statement:
```js
var foo = "foo", // more about foo
    bar = "bar"; // this comment causes the warning 

/* ... */
```

Perhaps we just need an option to allow trailing comments?

**Comment by btmills** [comment]
The current implementation uses a regex to look for newlines in the raw source code. Using tokens simplifies the logic greatly.

More specific to this issue, in my opinion, comments should be allowed at the end of the line by default, no option necessary. Thoughts?

Also, while I'm at it, the current error message is `Newline is {always/never} expected after a "var" statement.` Would it be a breaking change if the messages were altered? At the very least "var" should be "let" or "const" where appropriate, but I'd rather just change to "...after variable declarations." Further, this isn't just checking for a newline, but for a blank line. The naming seems misleading to me. So, pie in the sky proposal: deprecate `newline-after-var`, add a new rule, name TBD (`blank-line-after-variable-declaration` is way too long), same always/never option, whose messages are `Unexpected blank line after variable declarations.` and `Expected blank line after variable declarations.`

**Comment by cvisco** [comment]
@btmills: 100% Agree permitting trailing comments *should* be default. I was thinking at first someone might not want that, but I just realized there is a separate rule for dealing with trailing comments anyway: `no-inline-comments`.

**Comment by nzakas** [comment]
Agree, end-of-line comments should be fine by default.

Changing the message isn't a breaking change.

I'd keep the rule name as-is, as `var` is just short for "variable declaration", which applies equally to `let`, `const`, and `var`.

**Comment by cvisco** [comment]
I'm going to take a look at resolving this today, if there are no objections.

**Comment by nzakas** [comment]
Go for it! Please be sure to read over our contributor guidelines: http://eslint.org/docs/developer-guide/contributing.html
